File: docs/code-examples/segmentation_image_simple.cpp
// Create and log a segmentation image.

#include <rerun.hpp>

#include <algorithm> // std::fill_n
#include <vector>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_annotation_context_connections");
    rec.spawn().exit_on_failure();

    // Create a segmentation image
    const int HEIGHT = 8;
    const int WIDTH = 12;
    std::vector<uint8_t> data(WIDTH * HEIGHT, 0);
    for (auto y = 0; y < 4; ++y) {                                         // top half
        std::fill_n(data.begin() + y * WIDTH, 6, static_cast<uint8_t>(1)); // left half
    }
    for (auto y = 4; y < 8; ++y) {                                             // bottom half
        std::fill_n(data.begin() + y * WIDTH + 6, 6, static_cast<uint8_t>(2)); // right half
    }

    // create an annotation context to describe the classes
    rec.log_timeless(
        "/",
        rerun::AnnotationContext({
            rerun::AnnotationInfo(1, "red", rerun::Rgba32(255, 0, 0)),
            rerun::AnnotationInfo(2, "green", rerun::Rgba32(0, 255, 0)),
        })
    );

    rec.log("image", rerun::SegmentationImage({HEIGHT, WIDTH}, data));
}

File: docs/snippets/all/archetypes/segmentation_image_simple.py
"""Create and log a segmentation image."""

import numpy as np
import rerun as rr

# Create a segmentation image
image = np.zeros((8, 12), dtype=np.uint8)
image[0:4, 0:6] = 1
image[4:8, 6:12] = 2

rr.init("rerun_example_segmentation_image", spawn=True)

# Assign a label and color to each class
rr.log("/", rr.AnnotationContext([(1, "red", (255, 0, 0)), (2, "green", (0, 255, 0))]), static=True)

rr.log("image", rr.SegmentationImage(image))
